insights/terraform/projects/bfd/common/s3.tf

```diff
@@ -72,3 +72,22 @@ resource "aws_s3_bucket" "bfd-insights-bfd-app-logs" {
     mfa_delete = false
   }
 }
+
+# TODO: Replace the following when/if insights Terraform is merged with main Terraform
+resource "aws_s3_bucket_notification" "bucket_notifications" {
+  bucket = module.bucket.id
+
+  dynamic "lambda_function" {
+    for_each = data.aws_lambda_function.server_regression_glue_triggers
+
+    content {
+      events = [
+        "s3:ObjectCreated:*",
+      ]
+      filter_prefix       = "databases/bfd-insights-bfd-${lambda_function.key}/bfd_insights_bfd_${replace(lambda_function.key, "-", "_")}_server_regression/"
+      filter_suffix       = ".stats.json"
+      id                  = "bfd-${lambda_function.key}-server-regression-glue-trigger"
+      lambda_function_arn = lambda_function.value.arn
+    }
+  }
+}
```

Comment:
Yeah, this is the only resource being added to the BFD Insights Terraform. I agree with you, this probably _would_ make more sense to live next to the other `server-regression` Terraform, however this particular resource [cannot be declared more than once for a given S3 Bucket](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_notification):

> S3 Buckets only support a single notification configuration. Declaring multiple `aws_s3_bucket_notification` resources to the same S3 Bucket will cause a perpetual difference in configuration.

Due to this restriction, I felt it was better to keep the resource next to where the Bucket is declared in case new event notifications need to be added that are not related to `server-regression`.

Reply:
Ah, I see. I didn't know that, and that makes a lot of sense. Thanks for explaining!